File: src/main/java/de/crazydev22/resourcesftp/LockableFtpClient.java
package de.crazydev22.resourcesftp;

import org.gradle.internal.concurrent.Stoppable;

public interface LockableFtpClient extends Stoppable {
    FtpHost getHost();
    boolean isConnected();
    <T> T run(FtpClientTask<T> task);
    <T> Runnable runWithLock(FtpClientTask<T> task);
}

File: src/main/kotlin/de/crazydev22/resourcesftp/LockableFtpClient.kt
package de.crazydev22.resourcesftp

import org.apache.commons.net.ftp.FTPClient
import org.gradle.internal.concurrent.Stoppable

interface LockableFtpClient : Stoppable {
    val host: FtpHost
    val isConnected: Boolean
    fun <T> run(task: (FTPClient) -> T): T
}
